dozer-cli/src/config_helper.rs

```diff
@@ -67,16 +46,39 @@ pub fn combine_config(config_paths: Vec<String>) -> Result<Option<String>, Confi
     }
 }
 
-fn merge_yaml(
+pub fn add_file_content_to_config(
+    combined_yaml: &mut serde_yaml::Value,
+    name: &str,
+    content: String,
+) -> Result<(), ConfigCombineError> {
+    if name.contains(".yml") || name.contains(".yaml") {
+        let yaml: serde_yaml::Value = serde_yaml::from_str(&content)
+            .map_err(|e| ConfigCombineError::ParseYaml(name.to_string(), e))?;
+        merge_yaml(yaml, combined_yaml, false)?;
+    } else if name.contains(".sql") {
+        let yaml = serde_yaml::Value::Mapping(Mapping::from_iter([(
+            serde_yaml::Value::String("sql".into()),
+            serde_yaml::Value::String(";".to_string() + content.as_str()),
+        )]));
+        merge_yaml(yaml, combined_yaml, true)?;
+    } else {
+        warn!("Config file \"{name}\" extension not supported");
+    }
+
+    Ok(())
+}
+
+pub fn merge_yaml(
     from: serde_yaml::Value,
     to: &mut serde_yaml::Value,
+    join_strings: bool,
```

Comment:
This parameter is only for the branch `name.contains(".sql")`. We don't really need to call `merge_yaml` there. Just insert or append to the `sql` string.

Reply:
Sure. Refactored that part.